common/src/main/java/io/druid/timeline/LogicalSegment.java

```diff
@@ -19,8 +19,10 @@
 
 package io.druid.timeline;
 
+import io.druid.guice.annotations.PublicApi;
 import org.joda.time.Interval;
 
+@PublicApi
```

Comment:
QueryToolChest.filterSegments(..) returns an impl of this , so this needs to be an extension point.